<launch>
  <param name="use_sim_time" value="true"/>

  <!-- MAVROS + SITL -->
  <include file="$(find mavros)/launch/apm.launch">
    <arg name="fcu_url" value="udp://:14550@127.0.0.1:14557"/>
    <!--   здесь убрали publish_tf -->
  </include>

  <!-- Опционально: статический transform base_link→lidar_link -->
  <node pkg="tf" type="static_transform_publisher" name="base_to_lidar"
        args="0 0 0.2 0 0 0 base_link lidar_link 100"/>

  <!-- SLAM (gmapping) -->
  <node pkg="gmapping" type="slam_gmapping" name="slam_gmapping" output="screen">
    <remap from="scan" to="/scan"/>
    <param name="odom_frame" value="odom"/>
    <param name="base_frame" value="base_link"/>
    <param name="map_update_interval" value="5.0"/>
  </node>

  <!-- Teleop для управления -->
  <node pkg="teleop_twist_keyboard" type="teleop_twist_keyboard.py" name="teleop">
    <remap from="cmd_vel" to="/mavros/setpoint_velocity/cmd_vel_unstamped"/>
  </node>

  <!-- RViz -->
  <node pkg="rviz" type="rviz" name="rviz"
        args="-d $(find basic_nav)/rviz/basic_nav.rviz"/>
</launch>
